src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -826,14 +829,8 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
 {
   QgsExpressionContext context = QgsLayoutItem::createExpressionContext();
 
-  // We only want the last scope from the map's expression context, as this contains
-  // the map specific variables. We don't want the rest of the map's context, because that
-  // will contain duplicate global, project, layout, etc scopes.
-
   if ( mMap )
-  {
-    context.appendScope( mMap->createExpressionContext().popScope() );
-  }
+    context.appendScopes( mMap->createExpressionContext().takeScopes() );
```

Comment:
Is the removed comment (about duplicated scopes) no longer valid?

Reply:
I'll revert this change, I was simply unsure if the atlas scope was in, but in theory it should be already.